controllers/machinehealthcheck_targets.go

```diff
@@ -95,12 +99,15 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 
 	// machine has failed
 	if t.Machine.Status.FailureReason != nil {
+		conditions.MarkFalse(t.Machine, clusterv1.ConditionHealthCheckSucceeded, "FailureReason", clusterv1.ConditionSeverityWarning, "Reason: %v", t.Machine.Status.FailureReason)
 		logger.V(3).Info("Target is unhealthy", "reason", t.Machine.Status.FailureReason)
 		return true, time.Duration(0)
 	}
 
 	// the node does not exist
 	if t.nodeMissing {
+		logger.V(3).Info("Target is unhealthy: node is missing")
+		conditions.MarkFalse(t.Machine, clusterv1.ConditionHealthCheckSucceeded, "NodeMissing", clusterv1.ConditionSeverityWarning, "")
```

Comment:
I'm not sure NodeMissing is easy to understand for users, but I don't have a better idea now...
(plus, use a constant for condition reason)

Reply:
Yeah maybe something like "NodeNotFound"? I think users should be able to get that